pkg/clustermesh/operator/clustermesh.go

```diff
@@ -112,8 +113,13 @@ func newClusterMesh(lc cell.Lifecycle, params clusterMeshParams) (*clusterMesh,
 		ClusterInfo:         params.ClusterInfo,
 		RemoteClientFactory: params.RemoteClientFactory,
 		NewRemoteCluster:    cm.newRemoteCluster,
-		ServiceResolver:     params.ServiceResolver,
-		Metrics:             params.CommonMetrics,
+		Resolvers: func() (out []dial.Resolver) {
+			if params.ServiceResolver != nil {
+				out = append(out, params.ServiceResolver)
+			}
+			return out
+		}(),
```

Comment:
Shouldn't this be like that?

```suggestion
		Resolvers: []dial.Resolver{params.ServiceResolver}
```
Or this is meant to have the `ServiceBackendResolver` too?

Reply:
Woops actually nevermind this isn't the same as the function check for nil 🤦‍♂️

Reply:
Yeah, it is a bit annoying to have to check for nil here, the reason was basically to not have to change tests, as that's the only place where the resolver can be nil. And I haven't added an equivalent `ServiceBackendResolver` for the operator (it would need to be implemented differently, as we don't have the statedb tables there) is that it does not take part in the circular dependency, and it should be able to connect as soon as the agent reconciles the services anyhow.